fedbiomed/researcher/aggregators/scaffold.py

```diff
@@ -373,24 +381,19 @@ def set_nodes_learning_rate_after_training(
 
         n_model_layers = len(training_plan.get_model_params())
         for node_id in self._fds.node_ids():
-            lrs: List[float] = []
+            lrs: Dict[str, float] = {}
 
             if training_replies[n_round].get_index_from_node_id(node_id) is not None:
                 # get updated learning rate if provided...
                 node_idx: int = training_replies[n_round].get_index_from_node_id(node_id)
-                lrs += training_replies[n_round][node_idx]['optimizer_args'].get('lr')
+                lrs.update(training_replies[n_round][node_idx]['optimizer_args'].get('lr'))
```

Comment:
is there ever a case where the two conditions below are both true?
1) `training_replies[n_round].get_index_from_node_id(node_id) is not None`
2) `training_replies[n_round][node_idx]['optimizer_args']` does NOT contain the `lr` key

If yes, then this will fail because `update` cannot be called with a `None` argument. 
But I'm not familiar enough with the whole process to know whether this scenario can happen or not.